src/routes/data-router.ts

```diff
@@ -0,0 +1,69 @@
+import { Router } from "express";
+import { getRepository } from "typeorm";
+import { Todo } from "../entities/todo";
+
+const dataRouter = Router();
+
+dataRouter.get("/Tasks", async (req, res) => {
+  try {
+    const todoRepository = getRepository(Todo);
+    const todos = await todoRepository.find({ order: { id: "ASC" } });
+    res.json(todos);
+  } catch (error) {
+    console.error("Error executing database query:", error);
+    res.status(500).json({ error: "Internal server error" });
+  }
+});
+
+dataRouter.get("/lastID", async (req, res) => {
+  try {
+    const todoRepository = getRepository(Todo);
+    const todos = await todoRepository.find({ order: { id: "DESC" } });
+    res.json(todos[0].id);
+  } catch (error) {
+    console.error("Error executing database query:", error);
+    res.status(500).json({ error: "Internal server error" });
+  }
+});
+
+dataRouter.post("/Tasks", async (req, res) => {
+  try {
+    const { text } = req.body;
+    const todoRepository = getRepository(Todo);
+    const todo = todoRepository.create({
+      text,
+      isComplete: false,
+    });
+    await todoRepository.insert(todo);
+    res.json("task added!");
```

Comment:
אם אני מבין את התהליך שלך נכון, הפרונט שולח בקשת POST כדי לשמור את הtask החדש, ואז עושה בקשת get כדי לקבל את ה ID של ה task שנשמר.
במקום זה אפשר להחזיר את ה ID בתור ה response וזה חוסך בקשה.
אפילו לא צריך לעשות שתי בקשות ל-DB אם אתה משתמש ב save במקום insert, פונקציה של typeorm שגם מחזירה את ה entity שנשמר במלואו כולל ה ID
```suggestion
    const savedTask = await todoRepository.save(todo);
    res.json(savedTask.id);
```

Reply:
צודק זה סתם עוד בקשה שיניתי לsave